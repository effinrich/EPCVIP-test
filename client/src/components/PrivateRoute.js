import React from 'react'
import PropTypes from 'prop-types'
import { Redirect, Route } from 'react-router-dom'

const PrivateRoute = ({ component: Component, location, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      localStorage.getItem('auth') ? (
        <Component {...props} />
      ) : (
        <Redirect to={{ pathname: '/login', state: { from: location } }} />
      )
    }
  />
)

PrivateRoute.propTypes = {
  component: PropTypes.object,
  location: PropTypes.object
}

export default PrivateRoute
