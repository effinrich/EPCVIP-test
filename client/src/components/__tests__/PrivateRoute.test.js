import React from 'react'

import { render } from 'utils/testUtils/helperRtl'

import PrivateRoute from '../PrivateRoute'

describe('PrivateRoute Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<PrivateRoute />)
    expect(container).toMatchSnapshot()
  })

  it('should render child element', () => {
    const { container } = render(<PrivateRoute />)
    expect(container.firstChild).toBeDefined()
  })
})
