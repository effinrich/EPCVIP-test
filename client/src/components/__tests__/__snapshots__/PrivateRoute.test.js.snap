// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Component should render without crashing 1`] = `<div />`;
